rfcs/0005-fork-selection.md

```diff
@@ -0,0 +1,166 @@
+# Fork Selection
+ This RFC defines Solana's *Nakomoto Fork Selection* algorithm based on time locks. It satisfies the following properties:
+
+
+* A voter can eventually recover from voting on a fork that doesn't become the fork with the largest finality.
```

Comment:
Does finality have a size?

Reply:
@garious it is up to the node.  It could be configured that the voter will not increase a lockout past 256 unless that fork has more than 50% of the global stake.

Reply:
I've always measured finality in time. largest == longest? Or are you talking about the cost of rollback?

Reply:
@garious Yes.  In terms of economic finality.